.github/workflows/deploy_branches.yml

```diff
@@ -0,0 +1,53 @@
+# General notes on github actions: Note that both the working directory
+# and environment variables generally are not shared between steps.
+name: Deploy hazel
+on: [push]
+jobs:
+  Deploy:
+    runs-on: ubuntu-latest
+    steps:
+      # NOTE: position the below lines in the code between two steps
+      # and uncomment them to open an ssh connection at that point:
+      #- name: Debugging with ssh
+      #  uses: lhotari/action-upterm@v1
+      - name: Checkout the hazel repo on the current branch # STEP 1
+        uses: actions/checkout@v2
+        with:
+          path: source
+      - name: Add the name of the current branch to the environment as BRANCH_NAME
+        uses: nelonoel/branch-name@v1.0.1
+      - name: Retrieve the build environment if cached # STEP 2
+        id: opam-cache
+        uses: actions/cache@v2
+        with:
+          path: './source/.opam/'
+          key: ${{ runner.os }}-modules-${{ hashFiles('./source/opam.export') }}
+      - name: Install dependencies and build hazel # STEP 3
+        run: |
+          sudo apt --assume-yes install m4 opam
+          export OPAMYES=1
+          opam init --compiler=ocaml-base-compiler.4.08.1
+          eval $(opam env)
+          make deps 
+          make release
+        working-directory: ./source
+      - name: Checkout the website repo # STEP 4
+        uses: actions/checkout@v2
+        with:
+          repository: hazelgrove/build
+          token: ${{ secrets.ACCESS_TOKEN }}
+          path: server
+      - name: Clear any old build of this branch # STEP 5
+        run: if [ -d "${BRANCH_NAME}" ] ; then rm -rf "${BRANCH_NAME}" ; fi 
+        working-directory: ./server
+      - name: Copy in the newly built source # STEP 6
+        run: mkdir "./server/${BRANCH_NAME}" && cp -r "./source/_build/default/src/hazelweb/www"/* "./server/${BRANCH_NAME}"
+      - name : Commit to the website aka deploy # STEP 7
+        run: |
+          git config user.name github-deploy-action
+          git config user.email hazel-deploy@hazel.org
+          git add -A
+          git status
+          git commit -m "github-deploy-action"
```

Comment:
can we have the commit message be more descriptive here, e.g. say which branch was modified?

Reply:
done. well, we'll see if it works in 22 minutes

Reply:
yep, branch name is added to commits now